src/snakemake/scheduling/milp.py

```
@@ -0,0 +1,201 @@
+from collections import defaultdict
```

Ruff (0.12.2)

`collections.defaultdict` imported but unused

Remove unused import: `collections.defaultdict`

(F401)